### PR TITLE
Remove support for serializing signed integers [for VarInt support]

### PR DESCRIPTION
**Pull request by TrustHenry**
Our Serializer and Deserializer are used only by unsigned Integers. 
So delete unused int,long and Modifying the int used in unittest to uint

### CODE REVIEW

File: source/agora/common/Deserializer.d [hunk 1]
```diff
@@ -103,7 +103,7 @@ public void deserializePart (T)(ref T record, scope DeserializeDg dg)
 {
     import std.traits;
     OriginalType!T orig_val;
-    deserializePart(orig_val, dg);
+    deserializePart(cast(OriginalType!T) orig_val, dg);
```
Comment (AndrejMitrovic) [inline source/agora/common/Deserializer.d]: `orig_val` is already typed as `OriginalType!T`.

Comment (TrustHenry) [reply source/agora/common/Deserializer.d]: Error: none of the overloads of deserializePart are callable using argument types (int, ubyte[] delegate(ulong size) @safe), candidates are:
Is there a way to solve this error?

Comment (Geod24) [reply source/agora/common/Deserializer.d]: Don't use `OriginalType`. Pass the member directly.

Comment (AndrejMitrovic) [reply source/agora/common/Deserializer.d]: Explained IRL.

So the actual issue was that `NetworkState` was another enum but it defaulted to `int`. So I suggested to make it `NetworkState : ubyte` in a separate commit, and getting rid of this cast here.

**Comment by TrustHenry** [comment]
ready for review.

**Comment by AndrejMitrovic** [comment]
I've renamed the PR.

**Comment by AndrejMitrovic** [comment]
> ready for review.

I don't know about that. All the test systems failed.

**Comment by MichaelKim20** [comment]
Signed element does not need to be treated with VarInt. Why do you have to delete it?

**Comment by TrustHenry** [comment]
> > ready for review.
> 
> I don't know about that. All the test systems failed.

Fixed it

**Review by Geod24 — APPROVED** [review]

**Pull request merged**